docs/reverse_proxy.md

```diff
@@ -195,17 +193,23 @@ frontend https
   use_backend matrix if matrix-host matrix-path
 
 frontend matrix-federation
-  bind :::8448 v4v6 ssl crt /etc/ssl/haproxy/synapse.pem alpn h2,http/1.1
-  http-request set-header X-Forwarded-Proto https if { ssl_fc }
-  http-request set-header X-Forwarded-Proto http if !{ ssl_fc }
-  http-request set-header X-Forwarded-For %[src]
-
+  # Uncomment the line in the end to disable TLS <= v1.2
+  bind *:8448,[::]:8448 ssl crt /path/to/certificate.pem alpn h2,http/1.1 # ssl-min-ver TLSv1.2
+  
   default_backend matrix
 
 backend matrix
-  server matrix 127.0.0.1:8008
+  option forwardfor
```

Comment:
unfortunately this is not secure, due to https://github.com/matrix-org/synapse/issues/9471 (for which fixes are welcome!) - `option forwardfor` makes haproxy *append* an entry to any existing `X-Forwarded-For` header, whereas synapse uses the *first* entry in the header.